bundles/org.openhab.binding.bluetooth.bluez/src/main/java/org/openhab/binding/bluetooth/bluez/internal/BlueZBridgeHandler.java

```diff
@@ -0,0 +1,241 @@
+        if (addr != null) {
+            this.adapterAddress = new BluetoothAddress(addr.toUpperCase());
+        } else {
+            // If configuration does not contain adapter address to use, exit with error.
+            logger.info("Adapter MAC address not provided");
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "address not set");
+            return;
+        }
+
+        logger.debug("Creating BlueZ adapter with address '{}'", adapterAddress);
+
+        updateStatus(ThingStatus.UNKNOWN, ThingStatusDetail.CONFIGURATION_PENDING, "Initializing");
+
+        deviceManagerFactory.getPropertiesChangedHandler().addListener(this);
+
+        discoveryJob = scheduler.scheduleWithFixedDelay(this::initializeAndRefreshDevices, 5, 10, TimeUnit.SECONDS);
+    }
+
+    @Override
+    public void dispose() {
+        deviceManagerFactory.getPropertiesChangedHandler().removeListener(this);
+        logger.debug("Termination of DBus BlueZ handler");
+
+        Future<?> job = discoveryJob;
+        if (job != null) {
+            job.cancel(false);
+            discoveryJob = null;
+        }
+
+        BluetoothAdapter localAdatper = this.adapter;
+        if (localAdatper != null) {
+            localAdatper.stopDiscovery();
+            this.adapter = null;
+        }
+
+        super.dispose();
+    }
+
+    private @Nullable BluetoothAdapter prepareAdapter(DeviceManagerWrapper deviceManager) {
+        // next lets check if we can find our adapter in the manager.
+        BluetoothAdapter localAdapter = adapter;
+        if (localAdapter == null) {
+            BluetoothAddress localAddress = adapterAddress;
+            if (localAddress != null) {
+                localAdapter = adapter = deviceManager.getAdapter(localAddress);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "No adapter address provided");
+                return null;
+            }
+        }
+        if (localAdapter == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                    "Native adapter could not be found for address '" + adapterAddress + "'");
+            return null;
+        }
+        // now lets confirm that the adapter is powered
+        if (!localAdapter.isPowered()) {
+            localAdapter.setPowered(true);
+            // give the device some time to power on
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.DUTY_CYCLE,
+                    "Adapter is not powered, attempting to turn on...");
+            return null;
+        }
+
+        // now lets make sure that discovery is turned on
+        if (!Boolean.TRUE.equals(localAdapter.isDiscovering())) {
+            // we will check for devices next time around
+            localAdapter.startDiscovery();
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.DUTY_CYCLE, "Starting discovery");
```

Comment:
Why OFFLINE? If you can start discovery on it, it seems to be available, i.e. ONLINE?

Reply:
After looking at how `startDiscovery` was implemented, it looks like it already checks `isDiscovering` as part of it.
I'll go ahead and simplify this logic then.

Reply:
I implemented this months ago - but if my memory is good - if you turn it ONLINE immediately, the various things depending on it will immediately try to connect, which will fail because the devices needs to be discovered before it's possible to connect. But I might be wrong, and I'll retest it when we will be done if all changes !